doc/builders/images/dockertools.section.md

```diff
@@ -312,3 +312,31 @@ buildImage {
 ```
 
 Creating base files like `/etc/passwd` or `/etc/login.defs` is necessary for shadow-utils to manipulate users and groups.
+
+## fakeNss {#ssec-pkgs-dockerTools-fakeNss}
+
+If your primary goal is providing a basic skeleton for user lookups to work,
+and/or a lesser privileged user, adding `pkgs.fakeNss` to
+`build*Image.contents` might be the better choice than a custom script running
```

Comment:
I'm not convinced that it's better. The supposedly declarative merging of `contents` is a significant source of uncertainty with `dockerTools`. In fact, merging directories and files is non-trivial, so we ought to provide a well-specified and well-documented method for doing so. Shell commands provide that, even if it's hard to admit that `contents` was a failed design.

Reply:
in the case of `etc/passwd` already being present in another layer, I'd expect the merging (with default priority) to just fail (?).

Shell commands are way less declarative. The code in `examples.nix` before f7ee2706c2aaef359dd05b6910dcbfa6b3493dee, and the `useradd --system --gid nginx nginx` in there wasn't very reproducible - you're not exactly sure what user id `useradd` will pick.